From 13ee11011726d72a2577af7329c2377b495a477a Mon Sep 17 00:00:00 2001
From: Huang Rui <rui1.huang@intel.com>
Date: Mon, 3 Apr 2023 22:20:45 +0800
Subject: [ENABLE] Fix vulkan loading issue in R

Framework will not load vulkan from /vendor/lib[64]/hw, so move it to
/vendor/lib[64]. Also vulkan.intel.so expects sha1 build-id, but meson
system fallback it md5. So just force the build-id to sha1.
Signed-off-by: Marc Mao <marc.mao@intel.com>-
---
 android/Android.mk           | 2 +-
 src/intel/vulkan/meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/android/Android.mk b/android/Android.mk
index 14246ef244b..1d7ad0bf131 100644
--- a/android/Android.mk
+++ b/android/Android.mk
@@ -173,7 +173,7 @@ $(eval $(call mesa3d-lib,libGLESv2_mesa,.so.2,egl,MESA3D_LIBGLESV2_BIN))
 
 # Modules 'vulkan.{driver_name}', produces '/vendor/lib{64}/hw/vulkan.{driver_name}.so' HAL
 $(foreach driver,$(BOARD_MESA3D_VULKAN_DRIVERS), \
-    $(eval $(call mesa3d-lib,vulkan.$(MESA_VK_LIB_SUFFIX_$(driver)),.so.0,hw,MESA3D_VULKAN_$(driver)_BIN)))
+    $(eval $(call mesa3d-lib,vulkan.$(MESA_VK_LIB_SUFFIX_$(driver)),.so.0,,MESA3D_VULKAN_$(driver)_BIN)))
 
 ifneq ($(filter true, $(BOARD_MESA3D_BUILD_LIBGBM)),)
 # Modules 'libgbm', produces '/vendor/lib{64}/libgbm.so'
diff --git a/src/intel/vulkan/meson.build b/src/intel/vulkan/meson.build
index e1339797b2e..a8741ada2b1 100644
--- a/src/intel/vulkan/meson.build
+++ b/src/intel/vulkan/meson.build
@@ -242,7 +242,7 @@ libvulkan_intel = shared_library(
   ],
   c_args : anv_flags,
   gnu_symbol_visibility : 'hidden',
-  link_args : [vulkan_icd_link_args, ld_args_build_id, ld_args_bsymbolic, ld_args_gc_sections],
+  link_args : [vulkan_icd_link_args, '-Wl,--build-id=sha1', ld_args_bsymbolic, ld_args_gc_sections],
   link_depends : vulkan_icd_link_depends,
   install : true,
 )
-- 
2.25.1

